src/hotspot/share/oops/oop.hpp

```diff
@@ -307,6 +307,9 @@ class oopDesc {
 
   // for code generation
   static int mark_offset_in_bytes()      { return offset_of(oopDesc, _mark); }
+#ifdef _LP64
+  static int nklass_offset_in_bytes()    { return mark_offset_in_bytes() + markWord::klass_shift / 8; }
```

Comment:
Ew. Need to assert that `markWord::klass_shift` is actually a multiple of 8.

Reply:
Done that.